src/org/ods/services/BitbucketService.groovy

```diff
@@ -309,6 +309,50 @@ class BitbucketService {
         logger.debugClocked("buildstatus-${buildName}-${state}")
     }
 
+    /**
+     * Creates a code insight report in bitbucket via API.
+     * For further information visit https://developer.atlassian.com/server/bitbucket/how-tos/code-insights/
+     *
+     * @param result One of: PASS, FAIL
+     */
+    @SuppressWarnings('ParameterCount')
+    void createCodeInsightReport(String link, String repo, String gitCommit, String title, String details,
+                                 String result) {
+        withTokenCredentials { username, token ->
+            def payload = "{" +
+                "\"title\":\"${title}\"," +
+                "\"reporter\":\"OpenDevStack\"," +
+                "\"createdDate\":${System.currentTimeMillis()}," +
+                "\"details\":\"${details}\"," +
+                "\"result\":\"${result}\"," +
+                "\"data\": [" +
+                "{" +
+                "\"title\":\"Link\"," +
+                "\"value\":{\"linktext\":\"${link}\",\"href\":\"${link}\"}," +
+                "\"type\":\"LINK\"" +
+                "}" +
+                "]" +
+                "}"
+            try {
+                script.sh(
+                    label: 'Create Bitbucket Code Insight report via API',
+                    script: """curl \\
+                        --fail \\
+                        -sS \\
+                        --request PUT \\
+                        --header \"Authorization: Bearer ${token}\" \\
+                        --header \"Content-Type: application/json\" \\
+                        --data '${payload}' \\
+                        ${bitbucketUrl}/rest/insights/1.0/projects/${project}/\
+repos/${repo}/commits/${gitCommit}/reports/${gitCommit}"""
```

Comment:
I think the report key should be more specific, Atlassian says on this:
```
The report key should be a unique string chosen by the reporter and should be unique enough not to potentially clash with report keys from other reporters. We recommend using reverse DNS namespacing or a similar standard to ensure that collision is avoided.
```

See https://docs.atlassian.com/bitbucket-server/rest/7.13.0/bitbucket-code-insights-rest.html#idp9

Reply:
Well, now the key is the commit-SHA. Perhaps adding something like "aqua-report"? Eg: "1212121212-aqua-report"

Reply:
I think the key is meant to be something "stable" like `org.opendevstack.aquasec`. It's already connected to the commit anyway ...

Reply:
Modified to `org.opendevstack.aquasec`